components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/upload_files_timer.bal

```diff
@@ -66,14 +66,24 @@ function searchFilesToUpload() returns (error?) {
 }
 
 function timerTask() {
-    map<any> vals = getConfigMapValue(ANALYTICS);
-    boolean uploadFiles = <boolean>vals[FILE_UPLOAD_TASK];
-    analyticsUsername = <string>vals[USERNAME];
-    analyticsPassword = <string>vals[PASSWORD];
+    boolean uploadFiles = <boolean>getConfigBooleanValue(FILE_UPLOAD_ANALYTICS,FILE_UPLOAD_TASK,true);
+    analyticsUsername = <string>getConfigValue(FILE_UPLOAD_ANALYTICS,USERNAME,"admin");
+    analyticsPassword = <string>getConfigValue(FILE_UPLOAD_ANALYTICS,PASSWORD,"admin");
+    if (isOldAnalyticsEnalbed) {
+        //enables config reads for older versions
+        uploadFiles = <boolean>getConfigBooleanValue(OLD_FILE_UPLOAD_ANALYTICS,FILE_UPLOAD_TASK,true);
+        analyticsUsername = <string>getConfigValue(OLD_FILE_UPLOAD_ANALYTICS,USERNAME,"admin");
+        analyticsPassword = <string>getConfigValue(OLD_FILE_UPLOAD_ANALYTICS,PASSWORD,"admin");
+    }
```

Comment:
Do we still need to read config values from the new file if `isOldAnalyticsEnabled` is true?